version: "3.8"
services:
  login:
    build:
      context: ./Server
      dockerfile: dockerfiles/login.dockerfile
    ports:
      - "8001:8001"
    volumes:
      - ./Server:/app  
    environment:
      - PYTHONPATH=/app  
    networks:
      - fastapi-network  

  server:
    build:
      context: ./Server
      dockerfile: dockerfiles/server.dockerfile
    ports:
      - "8000:8000"
    volumes:
      - ./Server:/app  
    environment:
      - PYTHONPATH=/app  
    depends_on:
      - login  
    networks:
      - fastapi-network  

  consumer:
    build:
      context: ./Consumer  
      dockerfile: Consumer.dockerfile
    ports:
      - "2323:2323"
    volumes:
      - ./Consumer:/app  
    networks:
      - fastapi-network  
    command: ["python", "consumer.py"]

  nginx:
    image: nginx:alpine
    container_name: nginx
    ports:
      - "80:80" 
    volumes:
      - ./Server/nginx.conf:/etc/nginx/nginx.conf:ro
    depends_on:
      - login
      - server
      - react
      - consumer
    networks:
      - fastapi-network

  react:
    build:
      context: ./Frontend
      dockerfile: Frontend.dockerfile
    ports:
      - "5173:5173"
    volumes:
      - ./Frontend/src:/app/src
    environment:
      - NODE_ENV=development
    networks:
      - fastapi-network

  mongo:
    image: mongo:latest
    container_name: mongo
    ports:
      - "27017:27017"
    volumes:
      - mongo-data:/data/db  
    networks:
      - fastapi-network

  face_recognition:
    build:
      context: ./Face_Recognition
      dockerfile: face_recognition_service.dockerfile
    volumes:
      - ./Face_Recognition:/app
      - ./Fete_Jupani_Detectate:/app/Fete_Jupani_Detectate
      - ./Fete_Straine_Detectate:/app/Fete_Straine_Detectate
    networks:
      - fastapi-network
    depends_on:
      - server
    environment:
      - PYTHONUNBUFFERED=1

networks:
  fastapi-network:
    driver: bridge

volumes: 
  mongo-data:
  
  